src/status_im/contexts/settings/wallet/events.cljs

```diff
@@ -82,3 +88,81 @@
+
+(rf/reg-event-fx :wallet/make-keypairs-accounts-fully-operable make-keypairs-accounts-fully-operable)
+
+(rf/reg-fx :effects.connection-string/import-keypair
+ (fn [{:keys [key-uid sha3-pwd keypairs-key-uids connection-string on-success on-fail]}]
+   (let [config-map (.stringify js/JSON
+                                (clj->js
+                                 {:receiverConfig
+                                  {:loggedInKeyUid   key-uid
+                                   :keystorePath     ""
+                                   :password         (security/safe-unmask-data
+                                                      sha3-pwd)
+                                   :keypairsToImport keypairs-key-uids}}))]
+     (-> (native-module/input-connection-string-for-importing-keypairs-keystores
+          connection-string
+          config-map)
+         (promesa/then (fn [res]
+                         (let [error (when (syncing-events/extract-error res)
+                                       (str "generic-error: " res))]
+                           (if-not (some? error)
+                             (on-success)
+                             (on-fail error)))))
+         (promesa/catch #(log/error "error import-keypair/connection-string " %))))))
+
+(defn connection-string-for-import-keypair
+  [{:keys [db]} [{:keys [sha3-pwd keypairs-key-uids connection-string]}]]
+  (let [key-uid (get-in db [:profile/profile :key-uid])]
+    {:fx [[:effects.connection-string/import-keypair
+           {:key-uid           key-uid
+            :sha3-pwd          sha3-pwd
+            :keypairs-key-uids keypairs-key-uids
+            :connection-string connection-string
+            :on-success        #(rf/dispatch [:wallet/make-keypairs-accounts-fully-operable
+                                              keypairs-key-uids])
+            :on-fail           #(rf/dispatch [:toasts/upsert
+                                              {:type :negative
+                                               :text %}])}]]}))
+
+(rf/reg-event-fx :wallet/connection-string-for-import-keypair connection-string-for-import-keypair)
+
+(defn success-keypair-qr-scan
+  [_ [connection-string keypairs-key-uids]]
+  {:fx [(if (sync-utils/valid-connection-string? connection-string)
+          [:dispatch
+           [:standard-auth/authorize-with-password
+            {:blur?             true
+             :theme             :dark
+             :auth-button-label (i18n/label :t/confirm)
+             :on-auth-success   (fn [password]
+                                  (rf/dispatch [:hide-bottom-sheet])
+                                  (rf/dispatch
+                                   [:wallet/connection-string-for-import-keypair
+                                    {:connection-string connection-string
+                                     :keypairs-key-uids keypairs-key-uids
+                                     :sha3-pwd          password}]))}]]
```

Comment:
This is pretty cool, I didn't know I could invoke the standard auth flow this way 👍

Reply:
yes, there's standard auth events for authorizing explicitly with biometrics or password and one which decides based on the availability of biometrics which one to trigger

Reply:
yes, there's standard auth events for authorizing explicitly with biometrics or password and one which decides based on the availability of biometrics which one to trigger